tests/unit/test_cmr_search.py

```diff
@@ -18,193 +18,230 @@
-        query_response = get_granules(shortname=shortname,
-                                      collection_version=collection_version,
-                                      provider=provider,
-                                      cmr_env=CMR_UAT,
-                                      token='foo')
+        query_response = get_granules(
+            collection_shortname=self.collection_short_name,
+            collection_version=self.collection_version,
+            provider=self.provider,
+            cmr_env=CMR_UAT,
+            auth_header=self.bearer_token_header
+        )
 
         # Check if `get_granules` returns the CMR GranuleQuery output
         self.assertListEqual(query_response, granule_response)
 
-        # Check if the `GranuleQuery` object was
-        # instantiated with the correct environment, token, and
-        # expected page size
+        # Check if the `GranuleQuery` object was instantiated with the correct
+        # environment, Authorization header, and expected page size
         granule_query_mock.assert_called_once_with(mode=CMR_UAT)
-        granule_query_mock.return_value.bearer_token.assert_called_once_with
-        ('foo')
+        self.assertDictEqual(granule_query_mock.return_value.headers,
+                             {'Authorization': self.bearer_token_header})
         granule_query_mock.return_value.get.assert_called_once_with(10)
 
+    @patch('varinfo.cmr_search.GranuleQuery', spec=GranuleQuery)
+    def test_with_launcpad_token(self, granule_query_mock):
+        ''' Test when `get_granules` is called with an Authorization header
+            that contains a LaunchPad token (e.g., no 'Bearer ' prefix), that
+            header is propagated as expected to the `GranuleQuery` instance.
+
+        '''
+        granule_response = [
+            {
+                'links': [
+                    {'rel': 'http://esipfed.org/ns/fedsearch/1.1/s3#'},
+                    {'rel': 'http://esipfed.org/ns/fedsearch/1.1/data#'}
+                ]
+            }
+        ]
+
+        # Set return_value of `granule_query_mock` to `granule_response`
+        granule_query_mock.return_value.get.return_value = granule_response
+        query_response = get_granules(self.collection_concept_id,
+                                      cmr_env=CMR_UAT,
+                                      auth_header=self.launchpad_token_header)
+
+        # Check if `get_granules` returns the CMR GranuleQuery output
+        self.assertListEqual(query_response, granule_response)
+
+        # Check if the `GranuleQuery` object was instantiated with the correct
+        # environment, Authorization header, and expected page size
+        granule_query_mock.assert_called_once_with(mode=CMR_UAT)
+        self.assertDictEqual(granule_query_mock.return_value.headers,
+                             {'Authorization': self.launchpad_token_header})
+        granule_query_mock.return_value.get.assert_called_once_with(10)
```

Comment:
Is this doing anything different from the test up on L68-70? Or just explicitly testing that 'Bearer' doesn't need to be involved?

Reply:
Yeah - this is an almost identical test, just a different type of token (without 'Bearer'). Prior to this change there was some bearer token specific stuff that would try to take the input and prepend 'Bearer ' to it.

Reply:
Possibly this could be parameterised... will look into.

Reply:
...was thinking more of some of the bells and whistles that you get with `pytest`. Going to give in to laziness here.